### PR TITLE
Fix SE with  non FQDN hosts

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
/cc @GregHanson

### CODE REVIEW

File: pilot/pkg/proxy/envoy/v2/eds.go [hunk 1]
```diff
@@ -301,10 +301,15 @@ func (s *DiscoveryServer) updateServiceShards(push *model.PushContext) error {
...
 	svc2account := map[string]map[string]bool{}
 
-	for _, registry := range registries {
-		// Each registry acts as a shard - we don't want to combine them because some
-		// may individually update their endpoints incrementally
-		for _, svc := range push.Services(nil) {
+	// Each registry acts as a shard - we don't want to combine them because some
+	// may individually update their endpoints incrementally
+	for _, svc := range push.Services(nil) {
+		for _, registry := range registries {
```
Comment (hzxuzhonghu) [inline pilot/pkg/proxy/envoy/v2/eds.go]: I changed the order to reduce calling ` push.Services`

File: pilot/pkg/proxy/envoy/v2/eds.go [hunk 2]
```diff
@@ -301,10 +301,15 @@ func (s *DiscoveryServer) updateServiceShards(push *model.PushContext) error {
...
+			// in case this svc does not belong to the registry
+			if svc, _ := registry.GetService(svc.Hostname); svc == nil {
```
Comment (hzxuzhonghu) [inline pilot/pkg/proxy/envoy/v2/eds.go]: ignore the svc that not resides in this registry.

**Comment by howardjohn** [comment]
can you describe the issue, add a test, etc? No idea what this change does without context

**Comment by hzxuzhonghu** [comment]
@howardjohn FYI #13436

**Comment by hzxuzhonghu** [comment]
@howardjohn mind adding a label?

File: pilot/pkg/proxy/envoy/v2/eds.go (hunk 1 above)
Comment (howardjohn) [reply pilot/pkg/proxy/envoy/v2/eds.go]: Nice change

**Comment by hzxuzhonghu** [comment]
We should not depend on the `hostname` to get name/namespace, it is totally wrong.
For different registry, the hostname is different. And for serviceEntry, the hostname can be anything, like google, github, etc.

**Comment by linsun** [comment]
@hzxuzhonghu could we have a test for this?

Is this only for non http, https or TLS case?   for example, this test can use wikipedia as hosts instead of the current "www.wikipedia.org".

https://github.com/istio/istio/blob/master/tests/e2e/tests/pilot/testdata/networking/v1alpha3/service-entry-tcp-wikipedia-cidr.yaml#L14

**Comment by hzxuzhonghu** [comment]
>could we have a test for this?

Can do that.

>Is this only for non http, https or TLS case? 

No, it does not care about the protocol. It fits for all protocols.

**Pull request merged**